src/lib/support/IntrusiveList.h

```diff
@@ -414,6 +414,11 @@ class IntrusiveList : public IntrusiveListBase
         ConstIterator(IntrusiveListBase::ConstIteratorBase && base) : IntrusiveListBase::ConstIteratorBase(std::move(base)) {}
         const T * operator->() { return Hook::ToObject(mCurrent); }
         const T & operator*() { return *Hook::ToObject(mCurrent); }
+
+        ConstIterator & operator++() { return static_cast<ConstIterator &>(IntrusiveListBase::ConstIteratorBase::operator++()); }
+        ConstIterator operator++(int) { return IntrusiveListBase::ConstIteratorBase::operator++(1); }
```

Comment:
How come this and the `--(int)` version do not need static casts on the return type?  Same thing below.

Reply:
They return a new object rather than `this`, and that new object is actually of the base class.
Pretty sure this ends up using the `ConstIterator(IntrusiveListBase::ConstIteratorBase && base)` move constructor.